### ISSUE TITLE
Fix for `FAILED TO SEND AUTHENTICATION TO SERVER, no mechanism available` error

### ISSUE THREAD

**Issue by saschat** [lericson/pylibmc#246]
The current version of `libmemcached` has a [bug](https://bugs.launchpad.net/libmemcached/+bug/1381160) that results in the following `pylibmc` error:
```
MemcachedError: error 40 from memcached_get: FAILED TO SEND AUTHENTICATION TO SERVER, no mechanism available -> libmemcached/sasl.cc:238
```
which affects quite a lot of people, many of whom switch to [python-binary-memcached](https://github.com/jaysonsantos/python-binary-memcached).

Now this is a `libmemcached` problem. Unfortunately, `libmemcached` does not seem to be maintained any longer. For this reason I forked the `libmemcached` library [here](https://github.com/memcachier/libmemcached), fixed this bug, and added instructions on how to make sure `pylibmc` uses the fixed library. It is however unlikely that the fixed `libmemcached` will make into Ubuntu any time soon.

This is where `pylibmc` comes in. I noticed that as of 1.6.0 you ship a wheel file that contains `libmemcached.so`. Is it possible to release future versions of `pylibmc` with the patched `libmemcached.so`?

**Comment by saschat** [comment]
Hey, sorry for the late reply!

Heroku managed to get Canonical to fix the bug, so the latest version of `libmemcached` (1.0.18-4) is now fixed and available in Ubuntu's repositories. Could you release an updated version that contains the fixed `libmemached`? Thanks a lot!

**Comment by saschat** [comment]
@lericson is there anything blocking repackaging version 1.6.0 or releasing a newly packaged 1.6.1?

**Comment by lericson** [comment]
I’m pretty sure PyPI doesn’t allow replacing released files

**Comment by saschat** [comment]
Ok, then I guess it will have to be a version bump. Do you want a PR for a version bump?

**Comment by saschat** [comment]
@lericson Any update on this? It would be nice to have a `pylibmc` release with a `libmemcached` version that actually works for everybody.

**Comment by lericson** [comment]
So wait, sorry for coming back to this late, but libmemcached is basically abandoned?

**Comment by lericson** [comment]
By the way, your homemade "release" is not properly done. It needs to be autotoolized. I have painstakingly done this on the manywheel build now.

**Comment by saschat** [comment]
`libmemcached` has no active development but it is maintained by Canonical (the makers of Ubuntu). They fixed the bug and released version 1.0.18-4 at the request of Heroku, which has a lot of customers using `pylibmc`. 

My homemade "release"? I don't have one. I have been using `pylibmc` 1.5.2 with the Ubuntu shipped `libmemcached` until now.

**Comment by lericson** [comment]
This https://github.com/memcachier/libmemcached/releases

**Comment by saschat** [comment]
Ah, sorry, this repo does not work and is archived. I tried that bug fix before Canonical fixed it upstream and released it for all the Ubuntu versions. So the fixed `libmemcached` now ships with Ubuntu.

**Comment by lericson** [comment]
Yeah I got that, the problem is that to fix this for manylinux, I need to build the fixed version. It's turning out to be a lot more difficult than I'd like.

**Comment by lericson** [comment]
... build on CentOS in Docker, that is. Like building legos with five foot pincers.

**Comment by saschat** [comment]
I am sorry, I don't know much about that. I assumed you would just do the same you did to build the 1.6.0 wheel with this new release. How is it different to build 1.0.18-4 in comparison to 1.0.18-3?

**Comment by lericson** [comment]
1.0.18 is an actual release and so has ben libtoolized properly. I’m actually not sure what changed, right now it’s failing to run bison on libmemcached/csl/scanner.l.

**Comment by lericson** [comment]
This is now fixed, thanks for your time!

**Comment by saschat** [comment]
Thanks a ton!

**Comment by harem234** [comment]
i got this error with pylibmc==1.6.1
pylibmc.Error: error 40 from memcached_set: (0x9fb000) FAILED TO SEND AUTHENTICATION TO SERVER, no mechanism available,  host: 127.0.0.1:11211 -> libmemcached/sasl.cc:239

**Comment by chakmear** [comment]
> i got this error with pylibmc==1.6.1
> pylibmc.Error: error 40 from memcached_set: (0x9fb000) FAILED TO SEND AUTHENTICATION TO SERVER, no mechanism available, host: 127.0.0.1:11211 -> libmemcached/sasl.cc:239

I also have the error with 1.6.1, switching back to 1.5.2 solved it for me now...

**Comment by doihaveto** [comment]
If you're still experiencing this issue, try installing pylibmc without the binary:
`pip install pylibmc==1.6.1 --no-binary=:all:`

It will build pylibmc with the libmemcached version you have installed instead (make sure it's 1.0.18-4 with `dpkg -l | grep libmemcached-dev`).
